test/Builder/multi/different-environments.py

```diff
@@ -36,15 +36,15 @@
 
 _python_ = TestSCons._python_
 
-test.write('build.py', r"""#!/usr/bin/env python
+test.write('build.py', r"""#!/usr/bin/env python%s
 import sys
 def build(num, target, source):
     file = open(str(target), 'wb')
```

Comment:
Is there any reason for the file being open in binary mode?

Reply:
there are many places where files are opened using rb/wb and don't need to be. I assume somebody just did these out of habit. It's actually a poor idea to use binary mode for things that are supposed to be text files.

Reply:
Likely this was done as part of the py2 -> py2+py3 effort.
It may be that binary mode was needed to avoid Win32 line ending vs \n when comparing generated files for correctness by tests..
I remember I had to do that in some cases.

Reply:
> Likely this was done as part of the py2 -> py2+py3 effort.
> It may be that binary mode was needed to avoid Win32 line ending vs \n when comparing generated files for correctness by tests..
> I remember I had to do that in some cases.

Yup, the problem is if it's not consistent, and Py2 vs Py3 is unfortunately part of "not consistent".  I just ran into this and got some fails when I changed a test to not use binary mode when writing.  After digging, I see it fails because test.must_match() does default to binary mode.  Py2 does have the universal newlines mode, but it's deprecated since Py3 has a different scheme all around.

Reply:
Also you'll need to ensure changes work py3+2 on posix like and also windows.. The \n\r issues were part of why changes like those were made in tests..

Reply:
I have an experiment going which seems to indicate a lot of those \r\n (or alternatively/better os.linesep) things may not be needed as often.  Need to find some more time to work on it - and for Windows "full test run" not to be so slow... sigh.